CMakeLists.txt

```diff
@@ -7,6 +7,20 @@ if(Kokkos_ENABLE_CUDA)
   kokkos_check(OPTIONS CUDA_LAMBDA)
 endif()
 
+if(Kokkos_ENABLE_HIP AND NOT DEFINED GPU_TARGETS)
```

Comment:
Should we warn if `GPU_TARGETS` is defined and it does not match?

Reply:
Yes, I think that would be useful.